### PR TITLE
Feature/add cookies bithdtv

### PR DESCRIPTION
**Pull request by p0psicles**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

- [x] Streamlined the cookie based authentication by adding the required_cookies attribute. 
- [x] Display the required cookies in UI.
- [x] Add feedback through ui notifications, when cookies are not correct.
- [x] modified providers: Torrentday, Iptorrents, bithdtv.
- [x] Add wiki with screenshots with a step by step instruction.
- [x] Add link to wiki.
- [x] Open wiki link in a new tab.

### CODE REVIEW

File: medusa/providers/generic_provider.py [hunk 1]
```diff
@@ -726,27 +726,50 @@ def add_cookies_from_ui(self):
...
+
+        cookie_validator = re.compile(r'^([\w%]+=[\w%]+)(;[\w%]+=[\w%]+)*$')
+        if not cookie_validator.match(self.cookies):
+            ui.notifications.message(
+                'Failed to validate cookie for provider {provider}'.format(provider=self.name),
+                'Cookie is not correctly formatted: {0}'.format(self.cookies))
+            return {'result': False,
+                    'message': 'Cookie is not correctly formatted: {0}'.format(self.cookies)}
+
+        if not all(req_cookie in [x.rsplit('=', 1)[0] for x in self.cookies.split(';')] for req_cookie in self.required_cookies):
```
Comment (p0psicles) [inline medusa/providers/generic_provider.py]: We could potentially extend this, as we also know which cookies are missing. but that would require some additional code.

**Comment by p0psicles** [comment]
@duramato

File: medusa/providers/torrent/html/bithdtv.py [hunk 2]
```diff
@@ -190,12 +186,7 @@ def login(self):
             self.session.cookies.clear()
             return False
 
-        if '<h2>Login failed!</h2>' in response.text:
-            log.warning('Invalid username or password. Check your settings')
-            self.session.cookies.clear()
-            return False
-
-        return True
+        return self.validate_cookie_login(response, '<h2>Login failed!</h2>')
```
Comment (duramato) [inline medusa/providers/torrent/html/bithdtv.py]: Is the HTML markup really needed in the login failed string?

Comment (p0psicles) [inline medusa/providers/torrent/html/bithdtv.py]: No don't think so. I did that at first to make it more explicit. but I can use just something like Login Failed I asume.

**Comment by duramato** [comment]
Missing scenetime:

```
2017-08-15 02:37:02 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [SceneTime] :: [fa861a3] DailySearchQueueItem Exception, error: 'SceneTimeProvider' object has no attribute 'required_cookies'
Traceback (most recent call last):
  File "/home/pi/Medusa/medusa/search/queue.py", line 262, in run
    found_results = search_for_needed_episodes(force=self.force)
  File "/home/pi/Medusa/medusa/search/core.py", line 444, in search_for_needed_episodes
    cur_provider.cache.update_cache()
  File "/home/pi/Medusa/medusa/tv/cache.py", line 196, in update_cache
    data = self._get_rss_data()
  File "/home/pi/Medusa/medusa/tv/cache.py", line 179, in _get_rss_data
    return {'entries': self.provider.search(self.search_params)}
  File "/home/pi/Medusa/medusa/providers/torrent/html/scenetime.py", line 71, in search
    if not self.login():
  File "/home/pi/Medusa/medusa/providers/torrent/html/scenetime.py", line 190, in login
    self.add_cookies_from_ui()
  File "/home/pi/Medusa/medusa/providers/generic_provider.py", line 745, in add_cookies_from_ui
    if not all(req_cookie in [x.rsplit('=', 1)[0] for x in self.cookies.split(';')] for req_cookie in self.required_cookies):
AttributeError: 'SceneTimeProvider' object has no attribute 'required_cookies'
```

**Comment by duramato** [comment]
```
2017-08-15 13:48:10 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [IPTorrents] :: [76c9744] DailySearchQueueItem Exception, error: 'IPTorrentsProvider' object has no attribute 'validate_cookie_login'
Traceback (most recent call last):
  File "/home/pi/Medusa/medusa/search/queue.py", line 262, in run
    found_results = search_for_needed_episodes(force=self.force)
  File "/home/pi/Medusa/medusa/search/core.py", line 444, in search_for_needed_episodes
    cur_provider.cache.update_cache()
  File "/home/pi/Medusa/medusa/tv/cache.py", line 196, in update_cache
    data = self._get_rss_data()
  File "/home/pi/Medusa/medusa/tv/cache.py", line 179, in _get_rss_data
    return {'entries': self.provider.search(self.search_params)}
  File "/home/pi/Medusa/medusa/providers/torrent/html/iptorrents.py", line 65, in search
    if not self.login():
  File "/home/pi/Medusa/medusa/providers/torrent/html/iptorrents.py", line 178, in login
    return self.validate_cookie_login(response, 'Sign In')
AttributeError: 'IPTorrentsProvider' object has no attribute 'validate_cookie_login'
```

File: medusa/failed_processor.py [hunk 3]
```diff
@@ -43,11 +43,9 @@ def process(self):
...
             parsed = NameParser().parse(releaseName)
         except (InvalidNameException, InvalidShowException):
-            self.log(
-                logger.WARNING, u'Not enough information to parse release name into a valid show. '
-                u'Consider adding scene exceptions or improve naming for: {release}'.format
-                (release=releaseName)
-            )
+            self.log(logger.WARNING, u'Not enough information to parse release name into a valid show. '
+                     u'Consider adding scene exceptions or improve naming for: {release}'.format
+                     (release=releaseName))
```
Comment (fernandog) [inline medusa/failed_processor.py]: much better!

**Comment by p0psicles** [comment]
We shouldn't forget to remove bithdtv from broken providers list.

**Pull request merged**